langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java

```diff
@@ -13,18 +13,34 @@
  */
 public interface EmbeddingModel {
 
+    /**
+     * Embed a text.
+     *
+     * @param text the text to embed.
+     * @return the embedding.
+     */
     default Response<Embedding> embed(String text) {
         return embed(TextSegment.from(text));
     }
 
+    /**
+     * Embed a text segment.
+     *
+     * @param textSegment the text segment to embed.
+     * @return the embedding.
+     */
     default Response<Embedding> embed(TextSegment textSegment) {
         Response<List<Embedding>> response = embedAll(singletonList(textSegment));
-        return Response.from(
-                response.content().get(0),
-                response.tokenUsage(),
-                response.finishReason()
-        );
+        assert response.content().size() == 1 :
```

Comment:
Aren't assertions disabled by default? maybe better to use `ValidationUtils.ensureTrue(...)` or `ValidationUtils.ensureBetween(response.content().size(), 1, 1, "content")`

Reply:
Done